merge.go

```diff
@@ -515,3 +516,50 @@ func normalizeString(ctx context, opts *options, str string) (value, Error) {
 
 	return newSplice(ctx, opts.meta, varexp), nil
 }
+
+func fieldOptsOverride(opts *options, fieldName string) *options {
+	if opts.fieldValueHandling == nil {
+		return opts
+	}
+	cfgHandling, ok := opts.fieldValueHandling[fieldName]
+	if !ok {
+		// possible unknown depth defined
+		cfgHandling, ok = opts.fieldValueHandling[fmt.Sprintf("**.%s", fieldName)]
```

Comment:
I presume this is a hack for now? E.g. what about a pattern like this: `my.custom.path.**.paths`. 

EDIT: Hm, it looks like this case should be handled by `trimFieldConfigHandlingLevel`, although somewhat inefficient.


 In `path.go` you will find some path definitions and parsing. Maybe as a first step we can take the `fieldValueHandling` and build a prefix tree, that we can reuse (or do it more dynamically). In case we don't want to recompute the prefix-tree over and over again, we might provide some helpers to compile it once and pass the resulting tree as an option to go-ucfg.